src/v/cluster/cluster_utils.h

```diff
@@ -40,6 +40,23 @@ patch<broker_ptr> calculate_changed_brokers(
 
 /// Creates the same topic_result for all requests
 // clang-format off
+template<typename T>
```

Comment:
On top of Michals comment we also generally like to breakup PRs into smaller commits that make it easier for reviewers to review. How you break them up is up to you but we have a rule that at each commit in your PR, the branch must be at least buildable. Having all tests pass is a plus.

Moreover when you are checking in a commit that fixes an issue we use a special syntax in our commit messages that let github know this particular commit fixes a particular issue: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue this way its also checked into the github source history as well, and it will automatically close all linked issues when the PR is merged.